addons/web_editor/static/src/js/backend/field_html.js

```diff
@@ -497,6 +499,38 @@ var FieldHtml = basic_fields.DebouncedField.extend(TranslatableFieldMixin, {
             self._setValue(value);
         });
     },
+    /**
+     * Check stars on click event in readonly.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onReadonlyClickStar: function (ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+
+        const node = ev.target;
+        const previousStars = getAdjacentPreviousSiblings(node, sib => (
+            sib.nodeType === Node.ELEMENT_NODE && sib.className.includes('fa-star')
+        ));
+        const nextStars = getAdjacentNextSiblings(node, sib => (
+            sib.nodeType === Node.ELEMENT_NODE && sib.classList.contains('fa-star')
+        ));
+        const shouldToggleOff = node.classList.contains('fa-star') && !nextStars.length;
+        const rating = shouldToggleOff ? 0 : previousStars.length + 1
+
+        const starsId = parseInt($(node).parent().attr('id') || '0');
+        this._rpc({
+            route: '/web_editor/stars',
+            params: {
+                res_model: this.model,
+                res_id: this.res_id,
+                filename: this.name,
+                starsId,
+                rating,
+            },
+        }).then(value => this._setValue(value));
+    },
```

Comment:
It's a bit sad to have this logic somewhat duplicated with the editable version, though.

Reply:
Yes I know, I tried to reduce it as much as possible but couldn't do better than this.
The same was true for checklist by the way, except all that logic was duplicated in python rather than in js :p